optimum/onnxruntime/utils.py

```diff
@@ -271,3 +271,23 @@ class ORTQuantizableOperator(Enum):
     Resize = "Resize"
     AveragePool = "AveragePool"
     Concat = "Concat"
+
+
+def _get_external_data_paths(src_file_names, dst_file_names):
+    import onnx
+    from onnx.external_data_helper import ExternalDataInfo, _get_initializer_tensors
+
+    model_paths = src_file_names.copy()
```

Comment:
It's a list of Paths so I don't think it copies anything here, we might as well start from a new empty list, and fill it.

Reply:
I checked without the `copy()` and the extending does modify the list `src_file_names` unfortunately :/

Reply:
Yes, my point is that you can create an empty list

```python
model_paths = []
```

And fill it as you go?

My point here is that `src_files_names[0]` will be the same instance of `model_paths[0]`.

Reply:
I'm sorry I don't quite get it. I only use `model_paths` to iterate over the inputted `src_files_names`. Then I keep on filling `src_files_names`
```
    model_paths = []
    for model_path in model_paths:
        # load model graph
        model = onnx.load(str(model_path), load_external_data=False)
        # filter out tensors that are not external data
        model_tensors = _get_initializer_tensors(model)
        model_tensors_ext = [
            ExternalDataInfo(tensor).location
            for tensor in model_tensors
            if tensor.HasField("data_location") and tensor.data_location == onnx.TensorProto.EXTERNAL
        ]
        src_paths.extend([model_path.parent / tensor_name for tensor_name in model_tensors_ext])
        dst_file_names.extend(model_tensors_ext)
    return src_paths, dst_file_names
```
So this shouldn't work